public/app/plugins/panel/geomap/utils/getGeoMapStyle.ts

```diff
@@ -0,0 +1,32 @@
+import { Style, Stroke, Fill } from 'ol/style';
+import { ColorDimensionConfig } from 'app/features/dimensions';
+import { GeoJSONMapperRule } from './checkFeatureMatchesStyleRule';
+
+export interface GeoMapStyle {
+  shape?: string;
+  fill?: ColorDimensionConfig;
+  stroke?: ColorDimensionConfig;
+  strokeWidth?: number;
+  rule?: GeoJSONMapperRule;
+}
+
+/**
+ * Gets a geomap style based on fill, stroke, and stroke width
+ * @returns ol style
+ */
+export const getGeoMapStyle = (config: GeoMapStyle, property: any) => {
+  return new Style({
+    fill:
+      config.fill &&
+      new Fill({
+        color: `${config.fill}`,
```

Comment:
> config.fill

Is working right now because you are actually saving that as a string in the json model.  The `ColorDimensionConfig` class has a field/fixed value it will save.

With the much simpler PR: #39499, we can work with the real saved value :)